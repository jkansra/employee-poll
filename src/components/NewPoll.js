import { useState } from "react";
import { connect } from "react-redux";
import { handleAddPoll } from "../actions/questions";
import { useNavigate } from "react-router-dom";

const NewPoll = ({ dispatch }) => {
  const [optionOne, setOptionOne] = useState("");
  const [optionTwo, setOptionTwo] = useState("");
  const navigate = useNavigate();

  const handleOptionOneChange = (e) => {
    const optionOneText = e.target.value;

    setOptionOne(optionOneText);
  };

  const handleOptionTwoChange = (e) => {
    const optionTwoText = e.target.value;

    setOptionTwo(optionTwoText);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(handleAddPoll(optionOne, optionTwo));

    setOptionOne("");
    setOptionTwo("");

    navigate(`/home`)

  };

  return (
    <div className="mb-20">
      <h3 className="center">Would You Rather</h3>
      <p className="center">Create your own poll</p>
      {/* Form */}
      <form className="new-poll" onSubmit={handleSubmit}>
        <label className="center mb-10">First Option</label>
        <input placeholder="Option One" className="mb-20" value={optionOne} onChange={handleOptionOneChange} data-testid="first-option-input" />
        <label className="center mb-10">Second Option</label>
        <input placeholder="Option Two" className="mb-20" value={optionTwo} onChange={handleOptionTwoChange} data-testid="second-option-input" />
        <button className="center" type="submit" disabled={optionOne === "" || optionTwo === ""}>
          Submit
        </button>
      </form>
    </div>
  );
};

export default connect()(NewPoll);
