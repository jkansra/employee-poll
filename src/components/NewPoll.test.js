import { fireEvent, render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '..';
import NewPoll from './NewPoll';

describe("Snap <NewPoll />", () => {
    it("snap <NewPoll /> component successfully", () => {
        const component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <NewPoll />
                </Provider>
            </MemoryRouter>
        );
        expect(component).toMatchSnapshot()
    })

    it("Submit a new Poll successfully", () => {
        const component = render(
            <MemoryRouter>
                <Provider store={store}>
                    <NewPoll />
                </Provider>
            </MemoryRouter>
        );
        const { getByTestId, getByText } = component
        const firstOption = getByTestId("first-option-input");
        fireEvent.change(firstOption, { target: { value: "React" } })
        const secondOption = getByTestId("second-option-input");
        fireEvent.change(secondOption, { target: { value: "React-Redux" } })
        const submitBtn = getByText("Submit")
        fireEvent.click(submitBtn);
        expect(firstOption.value).toEqual("")
        expect(secondOption.value).toEqual("")
    })
})
