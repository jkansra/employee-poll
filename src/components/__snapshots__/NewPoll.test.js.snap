// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap <NewPoll /> snap <NewPoll /> component successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mb-20"
      >
        <h3
          class="center"
        >
          Would You Rather
        </h3>
        <p
          class="center"
        >
          Create your own poll
        </p>
        <form
          class="new-poll"
        >
          <label
            class="center mb-10"
          >
            First Option
          </label>
          <input
            class="mb-20"
            data-testid="first-option-input"
            placeholder="Option One"
            value=""
          />
          <label
            class="center mb-10"
          >
            Second Option
          </label>
          <input
            class="mb-20"
            data-testid="second-option-input"
            placeholder="Option Two"
            value=""
          />
          <button
            class="center"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mb-20"
    >
      <h3
        class="center"
      >
        Would You Rather
      </h3>
      <p
        class="center"
      >
        Create your own poll
      </p>
      <form
        class="new-poll"
      >
        <label
          class="center mb-10"
        >
          First Option
        </label>
        <input
          class="mb-20"
          data-testid="first-option-input"
          placeholder="Option One"
          value=""
        />
        <label
          class="center mb-10"
        >
          Second Option
        </label>
        <input
          class="mb-20"
          data-testid="second-option-input"
          placeholder="Option Two"
          value=""
        />
        <button
          class="center"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
